GliaWidgetsTests/Sources/ChatViewModel/ChatViewModelTests.swift

```diff
@@ -726,6 +726,90 @@ class ChatViewModelTests: XCTestCase {
         XCTAssertEqual(viewModel.messagesSection.items.count, 0)
         XCTAssertEqual(viewModel.receivedMessageIds, [])
     }
+
+    func test_viewModelDoesNotCallEndSessionWhenEngagementIsEndedByOperatorAndSurveyExists() throws {
+        enum Call { case finished }
+        var calls: [Call] = []
+        var interactorEnv = Interactor.Environment.mock
+        let data = try JSONSerialization.data(withJSONObject: [
```

Comment:
Let's create dedicated mock for Survey, to allow to use it easily for other unit tests.

Reply:
Added 61ca5dc73c5c9c63efcbbb8588d51f0baad5244d